dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java

```diff
@@ -76,40 +77,56 @@ private static List<String> filesExist(List<String> paths, String base) {
             }
             Path pathToFile = Paths.get(base, path);
             if (!Files.exists(pathToFile)) {
-                missingFiles.add(pathToFile.toString() + FILE_DOES_NOT_EXIST);
+                missingFiles.add(pathToFile + FILE_DOES_NOT_EXIST);
             }
         }
         return missingFiles;
     }
 
     // Determines if all the files in dockstoreYaml12 exist
     private static void allFilesExist(DockstoreYaml12 dockstoreYaml12, String basePath) throws ValidateYamlException {
+        List<String> filePathsToCheck = new ArrayList<>();
         List<String> missingFiles = new ArrayList<>();
 
         // Check Workflows
         List<YamlWorkflow> workflows = dockstoreYaml12.getWorkflows();
         if (!workflows.isEmpty()) {
             for (YamlWorkflow workflow : workflows) {
-                List<String> filePaths = workflow.getTestParameterFiles();
-                filePaths.add(workflow.getPrimaryDescriptorPath());
-                missingFiles.addAll(filesExist(filePaths, basePath));
+                List<String> testParameterFiles = workflow.getTestParameterFiles();
+
+                if (testParameterFiles != null) {
+                    filePathsToCheck.addAll(testParameterFiles);
+                }
+
+                filePathsToCheck.add(workflow.getPrimaryDescriptorPath());
             }
         }
         // Check Tools
         List<YamlTool> tools = dockstoreYaml12.getTools();
         if (!tools.isEmpty()) {
             for (YamlWorkflow tool : tools) {
-                List<String> filePaths = tool.getTestParameterFiles();
-                filePaths.add(tool.getPrimaryDescriptorPath());
-                missingFiles.addAll(filesExist(filePaths, basePath));
+                List<String> testParameterFiles = tool.getTestParameterFiles();
+
+                if (testParameterFiles != null) {
+                    filePathsToCheck.addAll(testParameterFiles);
+                }
+
+                filePathsToCheck.add(tool.getPrimaryDescriptorPath());
             }
         }
 
         // Check service
         Service12 service = dockstoreYaml12.getService();
         if (service != null) {
-            missingFiles.addAll(filesExist(service.getFiles(), basePath));
+            List<String> files = service.getFiles();
+            if (files != null) {
+                filePathsToCheck.addAll(files);
+            } else {
+                missingFiles.add(SERVICE_DOES_NOT_HAVE_FILES);
+            }
         }
+
+        missingFiles.addAll(filesExist(filePathsToCheck, basePath));
```

Comment:
Comment is for line 122/139, and it was like this, but `INVALID_FILE_STRUCTURE`, which translates to `Your file structure has the following errors:` -- I would say something like `Your .dockstore.yml has the following errors:` -- INVALID_FILE_STRUCTURE is slightly misleading variable name too (the structure of the file is fine in this case, it's just values that aren't good).